test/arrow_test.cc

```diff
@@ -59,4 +67,192 @@ TEST(ArrowCDataTest, CheckArrowSchemaAndArrayByNanoarrow) {
+                           .arrow_type = ::arrow::decimal128(10, 2)},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<DateType>(),
+                           .arrow_type = ::arrow::date32()},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<TimeType>(),
+                           .arrow_type = ::arrow::time64(arrow::TimeUnit::MICRO)},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<TimestampType>(),
+                           .arrow_type = ::arrow::timestamp(arrow::TimeUnit::MICRO)},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<TimestampType>(),
+                           .arrow_type = ::arrow::timestamp(arrow::TimeUnit::MICRO)},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<StringType>(),
+                           .arrow_type = ::arrow::utf8()},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<BinaryType>(),
+                           .arrow_type = ::arrow::binary()},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<UuidType>(),
+                           .arrow_type = ::arrow::extension::uuid()},
+        ToArrowSchemaParam{.iceberg_type = std::make_shared<FixedType>(20),
+                           .arrow_type = ::arrow::fixed_size_binary(20)}));
+
+namespace {
+
+void CheckArrowField(const ::arrow::Field& field, ::arrow::Type::type type_id,
+                     std::string_view name, bool nullable, int32_t field_id) {
+  ASSERT_EQ(field.name(), name);
+  ASSERT_EQ(field.nullable(), nullable);
+  ASSERT_EQ(field.type()->id(), type_id);
+
+  auto metadata = field.metadata();
+  ASSERT_TRUE(metadata != nullptr);
+  ASSERT_TRUE(metadata->Contains(kFieldIdKey));
+  ASSERT_EQ(metadata->Get(kFieldIdKey), std::to_string(field_id));
+}
+
+}  // namespace
+
+TEST(ToArrowSchemaTest, StructType) {
+  constexpr int32_t kStructFieldId = 1;
+  constexpr int32_t kIntFieldId = 2;
+  constexpr int32_t kStrFieldId = 3;
+
+  constexpr std::string_view kStructFieldName = "struct_field";
+  constexpr std::string_view kIntFieldName = "int_field";
+  constexpr std::string_view kStrFieldName = "str_field";
+
+  auto struct_type = std::make_shared<StructType>(std::vector<SchemaField>{
+      SchemaField::MakeRequired(kIntFieldId, std::string(kIntFieldName),
+                                std::make_shared<IntType>()),
+      SchemaField::MakeOptional(kStrFieldId, std::string(kStrFieldName),
+                                std::make_shared<StringType>())});
+  Schema schema(
+      /*schema_id=*/0, {SchemaField::MakeRequired(
+                           kStructFieldId, std::string(kStructFieldName), struct_type)});
+
+  ArrowSchema arrow_schema;
+  ASSERT_OK(ToArrowSchema(schema, &arrow_schema));
+
+  auto imported_schema = ::arrow::ImportSchema(&arrow_schema).ValueOrDie();
+  ASSERT_EQ(imported_schema->num_fields(), 1);
+
+  auto field = imported_schema->field(0);
+  CheckArrowField(*field, ::arrow::Type::STRUCT, kStructFieldName, /*nullable=*/false,
```

Comment:
technically you should `ASSERT_NO_FATAL_FAILURE(CheckArrowField(...))`

Reply:
You're right. I forgot to add them.